controllers/druid/druid_client.go

```diff
@@ -0,0 +1,76 @@
+package druid
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type DruidClient struct {
+	BaseUrl  string
+	UserName string
+	PassWord string
+	client   *http.Client
+}
+
+func InitClient(BaseUrl string, UserName string, PassWord string) DruidClient {
+	d := DruidClient{BaseUrl, UserName, PassWord, &http.Client{}}
+	return d
+}
+
+func (d DruidClient) GetCoordinatorConfig() (map[string]interface{}, error) {
+	req, err := http.NewRequest("GET", d.BaseUrl+"/druid/coordinator/v1/config", nil)
+	req.SetBasicAuth(d.UserName, d.PassWord)
+	resp, err := d.client.Do(req)
```

Comment:
Might need to handle non 200 status code responses. Not sure if they would be captured in err.